### DISCUSSION TITLE
Improve documentation and error reporting for lack of networking in Pyodide and PyScript

### DISCUSSION THREAD

**Discussion by tiran** [pyscript/pyscript#1196] [Proposals]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a duplication
- [X] I already searched in Google and didn't find any good information or help

### What is the issue/comment/problem?

It appears to me that lots of users are surprised by the fact that networked-based Python programs do not work in [Pyodide](https://pyodide.org/en/stable/project/roadmap.html#write-http-client-in-terms-of-web-apis) and PyScript. They try to use Python packages like requests or database connectors, and run into all sorts of errors that are hard to understands. Missing `_ssl` extension is a common problem.

However it is a red herring for a much bigger issue. Networking in Emscripten does not work like on Linux or Windows-like systems. It's not just the fact that blocking sockets are not supported, because they would block the browser loop. Browser also do not permit raw TCP and UDP sockets due to their potential for abuse. The [direct socket](https://github.com/WICG/direct-sockets/blob/main/docs/explainer.md) proposal may make some forms of raw connections possible in the future.

In the mean time PyScript and Pyodide should provide documentation and better error reporting (if possible). It would spare users from running into the problem over and over again and reduce frustration.

**Comment by tiran** [comment]
- https://github.com/pyscript/pyscript/issues/91
- https://github.com/pyscript/pyscript/issues/162
- https://github.com/pyscript/pyscript/issues/413
- https://github.com/pyscript/pyscript/issues/524

**Comment by marimeireles** [comment]
That's a great point @tiran, thanks for exposing it and marking the relevant issues here.
I can't tackle it right now, but if you or anyone else would like to pick it up, it'd be awesome =)
